scripts/hid_configurator/gui/gui.py

```diff
@@ -142,13 +172,31 @@ def dfu(self, update_progressbar):
         else:
             info = 'DFU transfer failed'
             message = 'Restart application and try again'
-        print(info)
+
         dfu_label.text = info
         notification.notify(app_name='GUI Configurator', app_icon='nordic.ico', title=info, message=message)
 
-    def perform_fwreboot(self):
-        print('Performing fwreboot, please wait')
-        self.device.perform_fwreboot()
+    def dfu_fwreboot(self, update_reboot_animation):
+        dfu_label = self.root.ids.dfu_label
+        progression_label = self.root.ids.dfu_buttons.ids.progression_label
+        dfu_label.text = 'Image swap in progress'
```

Comment:
Is it visible at all? Before end of the function dfu_label.text is overwritten.

Reply:
Yes, `device.perform_dfu_fwreboot(update_reboot_animation)` takes a while (thread waits for device to complete firmware swap).